sunpy/net/scraper.py

```diff
@@ -80,24 +99,77 @@ class Scraper:
     pattern looks with the actual time.
     """
 
-    def __init__(self, pattern, regex=False, **kwargs):
-        if regex:
+    def __init__(self, format=None, pattern=None, regex=False, **kwargs):
```

Comment:
Sorry, I was wrong in my previous comment about this. I will blame covid and not my own incompetence.

We want to keep the API the same way around as it will break all the older clients. 

I haven't thought through what I am about to say but I think we would need to do this:

"pattern", "regex "are the first two arguments to the init, then it will be "format" for the new behavior. 

We check if pattern/regex are passed in, we issue a deprecation warning saying to use format keyword and provide a message with a link to a migration guide.

I would also make the format keyword to be required to be a keyword as a temporary workaround. TO avoid any API ambiguity.

Reply:
> I would also make the format keyword to be required to be a keyword as a temporary workaround.
ooh do you mean having the format keyword as a required argument? because wouldn't that also break all the older clients?

also wish you a fast and comfy recovery from covid!! and thanks for all the reviews, health issues have also got to me these days, i should be able to address them over the weekend

Reply:
> > I would also make the format keyword to be required to be a keyword as a temporary workaround.
> 
> do you mean having the format keyword as a required argument? because wouldn't that also break all the older clients?

No. I mean the use of `*` in the signature. 

`def __init__(self, pattern=None, regex=False, *, format=None, **kwargs):`

> also wish you a fast and comfy recovery from covid!! and thanks for all the reviews, health issues have also got to me these days, i should be able to address them all over the weekend.

Make sure to look after yourself!